include/vinecopulib/bicop/implementation/class.ipp

```diff
@@ -462,17 +511,80 @@ inline void
 Bicop::set_rotation(const int rotation)
 {
   check_rotation(rotation);
+  if ((rotation_ - rotation % 180) != 0) {
+    flip_var_types();
+  }
   rotation_ = rotation;
   bicop_->set_loglik();
 }
 
-//! set the parameters
+inline void
+Bicop::check_data(const Eigen::MatrixXd& u) const
+{
+  check_data_dim(u);
+  tools_eigen::check_if_in_unit_cube(u);
+}
+
+inline void
+Bicop::check_data_dim(const Eigen::MatrixXd& u) const
+{
+  size_t n_cols = u.cols();
+  size_t n_cols_exp = 2 + get_n_discrete();
+  if ((n_cols != n_cols_exp) & (n_cols != 4)) {
+    std::stringstream msg;
+    msg << "data has wrong number of columns; "
+        << "expected: " << n_cols_exp << " or 4, actual: " << n_cols 
+        << " (model contains ";
+    if (n_cols_exp == 2) {
+      msg << "no ";
+    } else {
+      msg << get_n_discrete() << " ";
+    }
+    msg << "discrete variables)." << std::endl;
```

Comment:
I'm going to be annoying by pointing out that it could print `(model contains 1 discrete variables)`, which is grammatically incorrect :) JK, I don't care.